foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java

```diff
@@ -74,7 +74,9 @@ void apiVersion() {
     stubFor(post(anyUrl()).willReturn(okJson("{}")));
     Try.of(() -> client.chatCompletion(new OpenAiChatCompletionParameters()));
 
-    verify(exactly(1), postRequestedFor(anyUrl()).withoutQueryParam("api-version"));
+    verify(
+        exactly(1),
+        postRequestedFor(anyUrl()).withQueryParam("api-version", equalTo("2024-02-01")));
```

Comment:
(Explanation)

Without the `api-version` we experienced _resource not found_ error messages from AI Core.

Reply:
Correct, this was intended. The idea being that it is possible for users to prevent the api version from being set, e.g. for cases where the some model works without api version, or maybe it might be set automatically by long access; or if the key of that header ever changes.

That is why the javadoc of the method explicitly states to call withApiVersion afterwards 


But to be fair, the above are rather hypothetical scenarios and i think it may be worthwhile to change it in favor of the more convenient approach of setting this automatically